automated install/basic-install.sh

```diff
@@ -1023,106 +1170,122 @@ setDNS() {
+        "${r}" "${c}"
+
+    result=$?
+    case ${result} in
+        "${DIALOG_OK}")
+            # If they chose yes,
+            printf " %b Installing Admin Web Interface\\n" "${INFO}"
+            # Set the flag to install the web interface
             INSTALL_WEB_INTERFACE=true
             ;;
-        Off)
-            printf "  %b Web Interface Off\\n" "${INFO}"
-            # or false
+        "${DIALOG_CANCEL}")
+            # If they chose no,
+            printf " %b Not installing Admin Web Interface\\n" "${INFO}"
+            # Set the flag to not install the web interface
             INSTALL_WEB_INTERFACE=false
-            # Deselect the web server as well, since it is obsolete then
             INSTALL_WEB_SERVER=false
             ;;
+        "${DIALOG_ESC}")
+            # User pressed <ESC>
+            printf " %bEscape pressed, exiting installer at Admin Web Interface choice.%b\\n" "${COL_LIGHT_RED}" "${COL_NC}"
+            exit 1
+            ;;
     esac
 
     # If the user wants to install the Web admin interface (i.e. it has not been deselected above)
-    if [[ "${INSTALL_WEB_SERVER}" == true ]]; then
+    if [[ "${INSTALL_WEB_INTERFACE}" == true ]]; then
         # Get list of required PHP modules, excluding base package (common) and handler (cgi)
         local i php_modules
         for i in "${PIHOLE_WEB_DEPS[@]}"; do [[ $i == 'php'* && $i != *'-common' && $i != *'-cgi' ]] && php_modules+=" ${i#*-}"; done
-        WebToggleCommand=(whiptail --separate-output --radiolist "Do you wish to install the web server (lighttpd) and required PHP modules?\\n\\nNB: If you disable this, and, do not have an existing web server and required PHP modules (${php_modules# }) installed, the web interface will not function. Additionally the web server user needs to be member of the \"pihole\" group for full functionality." "${r}" "${c}" 6)
-        # Enable as default and recommended option
-        WebChooseOptions=("On (Recommended)" "" on
-            Off "" off)
-        WebChoices=$("${WebToggleCommand[@]}" "${WebChooseOptions[@]}" 2>&1 >/dev/tty) || (printf "  %bCancel was selected, exiting installer%b\\n" "${COL_LIGHT_RED}" "${COL_NC}" && exit 1)
-        # Depending on their choice
-        case ${WebChoices} in
-            "On (Recommended)")
-                printf "  %b Web Server On\\n" "${INFO}"
-                # set it to true, as clearly seen below.
+        dialog --no-shadow --clear \
+            --backtitle "Pi-hole Installation" \
+            --title "Web Server" \
+            --yesno "\\n\\nA web server is required for the Admin Web Interface.\
+\\n\\nDo you want to install lighttpd and the required PHP modules?\
+\\n\\nNB: If you disable this, and, do not have an existing web server\
+and required PHP modules (${php_modules# }) installed, the web interface\
+will not function. Additionally the web server user needs to be member of\
+the \"pihole\" group for full functionality." \
```

Comment:
```sh
\\n\\nNB: If you disable this, and, do not have an existing web server \
and required PHP modules (${php_modules# }) installed, the web interface \
will not function. Additionally the web server user needs to be member of \
the \"pihole\" group for full functionality." \
```

Add a space at the end of each line (before `\`) to avoid 2 words collapsing into one.

Reply:
Making some advertising for the `suggestion` code fences here, which make it very transparent and easy to merge suggested changes 🙂:
```suggestion
\\n\\nNB: If you disable this, and, do not have an existing web server \
and required PHP modules (${php_modules# }) installed, the web interface \
will not function. Additionally the web server user needs to be member of \
the \"pihole\" group for full functionality." \
```